pkg/generator/helper/helper.go

```diff
@@ -197,7 +197,7 @@ func {{.StepName}}(config {{ .StepName }}Options, telemetryData *telemetry.Custo
 	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
 	err := run{{.StepName | title}}(&config, telemetryData, &c{{ range $notused, $oRes := .OutputResources}}, &{{ index $oRes "name" }}{{ end }})
 	if err != nil {
-		log.Entry().WithError(err).Fatal("step execution failed")
+		log.Entry().WithError(err).Fatalf("step execution failed: %s", err.Error())
```

Comment:
@stippi2 suggested in another PR to use `%w` for error logging. But I don't know how it then looks like in the log message.

Reply:
From the [docu](https://golang.org/pkg/fmt/)

```
If the format specifier includes a %w verb with an error operand, the returned error will implement an Unwrap method returning the operand. It is invalid to include more than one %w verb or to supply it with an operand that does not implement the error interface. The %w verb is otherwise a synonym for %v.
```

> Maybe we should adjust this in the formatter.

Where do we have to adjust that? Is is possible to customise this? In that case we can't rely on `WithError(./.)` maybe since someone might customise the error away. In that case we would not see the reason for the error. This in turn would be actually bad since that makes troubleshooting more complicate resulting in more cycles until an error is resolved.

Wouldn't it be better to have the error details `always` in the error message?

Reply:
No, not for logging, but when returning an error. So instead of

```
return fmt.Errorf("something happened: %v", err)
```

you should use

```
return fmt.Errorf("something happened: %w", err)
```

You will then return an error which can be unwrapped. Or more precisely, you don't break the wrapping chain, as "err" itself may already included wrapped errors, which would be lost, if not using `%w` or wrapping explicitly via `errors.Wrapf()`.

Reply:
OK, so as far as I understand the comment from @stippi2  we should not use `%w` _inside log messages_, right? `%w` should only be used when creating a new `Error` based on an existing one (... wrapping).

Reply:
Yes.